### DISCUSSION TITLE
Best practice to receive notifications from newly registered devices (cameras, doorbots..etc)?

### DISCUSSION THREAD

**Discussion by PeterBunBun** [dgreif/ring#1517] [Q&A] [answered]
Hi community, while we're subscribing to notifications, it seems like with this API is promoting subscription to each `device` instead of the whole account. Using `RingCamera.onNewNotification(notificationHandler)`

Am I understand this correctly? Or is there other way of subscribing that I missed?

And with this approach, we can't seem to receive notifications from camera that has just been registered/added to the account after this API's listener was spun up. And we have to restart the whole listener to include this new camera.

Has anyone hit this as well?

**Comment by tsightler** [comment, marked as answer]
I would say that the overall goal of this API isn't about receiving account wide notifications, it's to provide an API representation for the state of each device.  Currently the code parses out only notifications required to update device state for things like motion/dings, even though there are many other push notifications.

You can subscribe directly to the pushReceiver `onNotification` method and do what you will with the raw notifications (I do the same for `onDataUpdate` for locations in my own project), but this probably will not resolve any issue with not receiving push notifications for new devices because devices must be explicitly subscribed and, currently, there is no dynamic device discovery which means, as you point out, devices are only discovered and subscribed during initial startup.

There is an older PR, which would likely need to be updated for current versions, to add support for manually refreshing devices via a `refresh()` function, but, quite honestly, there has been little demand for this as most people are not running around adding a bunch of devices every day so most of the time the environment is fairly static.

**Reply by PeterBunBun** [reply]
Gotcha! Really good know, thanks!

I took a look at `pushReceiver.onNotification`, I found that this was being setup/configured within a private method `registerPushReceiver` in `RingApi`. And since the `pushReceiver` scope is well limited within this method, it's not easily accessible. I'm not sure if it's a good idea to initiate a new pushReceiver instance with the same token outside of `RingApi`, since all the `hid` and relative information are wrapped in the credentials, subscribing with the same fcm token and same hid at separate instance at the same time doesn't seem right (in my test, it sometimes break the subscription if the connection last longer)

I'm wondering if there's a chance that we can expose either `pushReceiver` or a new Observable that is piped from `pushReceiver.onNotification` so that we can access raw notification at `RingApi` level?

[And please pardon me for my rookie level understanding to Typescript]
